WalletWasabi.Daemon/Rpc/WasabiJsonRpcService.cs

```diff
@@ -263,6 +264,74 @@ public string PayInCoinJoin(BitcoinAddress address, Money amount)
 		return activeWallet.AddCoinJoinPayment(address, amount);
 	}
 
+	[JsonRpcMethod("listpaymentsincoinjoin")]
+	public JsonRpcResultList ListPaymentsInCoinJoin()
+	{
+		var activeWallet = Guard.NotNull(nameof(ActiveWallet), ActiveWallet);
+		AssertWalletIsLoaded();
+		var payments = activeWallet.BatchedPayments.GetPayments();
+		return payments.Select(x =>
+		{
+			var paymentResult = new JsonRpcResult
+			{
+				["id"] = x.Id,
+				["amount"] = x.Amount.Satoshi,
+				["destination"] = x.Destination.ScriptPubKey.ToHex()
+			};
+
+			var state = x.State;
+			var stateHistory = new List<JsonRpcResult>();
+			while (state != null)
```

Comment:
Looks like we need here if, instead of while.

Reply:
No. Try to understand the code.